### ISSUE TITLE
Consistent snapshot failure after upgrade to 2.0.1.18_canary

### ISSUE THREAD

**Issue by NickPyz** [duplicati/duplicati#1883]
I deleted 2.0.1.17 and replaced with 2.0.1.18 leaving the existing configuration / sqlite databases in place. Now I get the following warning after every scheduled job:
----------------------------------------------------------------------------------------------
Warnings: [Failed to create a snapshot: System.IO.FileLoadException: Could not load file or assembly 'AlphaFS, Version=2.0.0.0, Culture=neutral, PublicKeyToken=4d31a58f7d7ad5c9' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
-----------------------------------------------------------------------------------------------

OS: Windows 10 - 64 bit
Thanks.

**Comment by NickPyz** [comment]
Additional info:

(1)Windows VSS Service is set at 'automatic' - and is running
(2) Duplicati 2 running as administrator
(2) Affected jobs have snapshot-policy=On

**Comment by CyBiS** [comment]
I have the same issue (running on Windows 7 64 bit). The same error occurs also when trying to restore a backup.

**Comment by agrajaghh** [comment]
Same for me running Win10 64bit. With `snapshot-policy=off` the job is running without any errors.

**Comment by kenkendk** [comment]
This is likely caused by the update for AlphaVSS after switching to NuGet.
You can try copying the alphavss folder over from a previous version to see if it gets picked up.

**Comment by agrajaghh** [comment]
@kenkendk copying the alphavss folder from 2.0.1.17 doesn't work for me:
```
Fatal error
System.IO.FileLoadException: Die Datei oder Assembly "AlphaFS, Version=2.0.0.0, Culture=neutral, PublicKeyToken=4d31a58f7d7ad5c9" oder eine Abhängigkeit davon wurde nicht gefunden. Die gefundene Manifestdefinition der Assembly stimmt nicht mit dem Assemblyverweis überein. (Ausnahme von HRESULT: 0x80131040)

Dateiname: "AlphaFS, Version=2.0.0.0, Culture=neutral, PublicKeyToken=4d31a58f7d7ad5c9"
   bei Duplicati.Library.Snapshots.WindowsSnapshot..ctor(String[] sourcepaths, Dictionary`2 options)
   bei Duplicati.Library.Snapshots.SnapshotUtility.CreateSnapshot(String[] folders, Dictionary`2 options)
   bei Duplicati.Library.Main.Operation.BackupHandler.GetSnapshot(String[] sources, Options options, ILogWriter log)
   bei Duplicati.Library.Main.Operation.BackupHandler.Run(String[] sources, IFilter filter)

WRN: Protokollierung der Assemblybindung ist AUS.
Sie können die Protokollierung der Assemblybindungsfehler aktivieren, indem Sie den Registrierungswert [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) auf 1 festlegen.
Hinweis: Die Protokollierung der Assemblybindungsfehler führt zu einer gewissen Leistungseinbuße.
Sie können dieses Feature deaktivieren, indem Sie den Registrierungswert [HKLM\Software\Microsoft\Fusion!EnableLog] entfernen.```

**Comment by NickPyz** [comment]
copying alphavss folder from 2.0.1.17 into 2.0.1.18 doesn't work for me either and VSS snapshot fails.

**Comment by mnaiman** [comment]
Problem is in AlphaVSS but in Aplha FS. There is some mismatch in dependencies, because Duplicati installs 1.5 but is dependant on 2.0 AlphaFS from here https://github.com/alphaleonis/AlphaFS/releases

**Comment by mnaiman** [comment]
Replace AlphaFS.dll and AlphaFS.xml from 2.0 release and error is gone.

**Comment by mr-flibble** [comment]
mnaiman: Can you please link 2.0 release that works for you? Thanks

Copying AlphaFS.dll and AlphaFS.xml from 2.0.1.15  doesn't work for me

**Comment by mnaiman** [comment]
https://github.com/alphaleonis/AlphaFS/releases/download/2.0/AlphaFS-2.0.0.zip

This is working. But if you have 2.0.1.18 in update folder you have to move it to main folder and delete update folder, because in update folder checksumming is protecting integrity of files.

**Comment by mr-flibble** [comment]
**Thank you for solution!** Copying AlphaFS.dll and AlphaFS.xml from net451 folder in https://github.com/alphaleonis/AlphaFS/releases/download/2.0/AlphaFS-2.0.0.zip works. No error message and backup works again.

**Comment by NickPyz** [comment]
This solution didn't work for me. I moved the 2 files from the extracted net451 folder into the 'duplicati 2' executable folder, but I get a snapshot failure: 

_Warnings: [Failed to create a snapshot: System.IO.FileNotFoundException: The system cannot find the file specified. (Exception from HRESULT: 0x80070002)
   at System.Reflection.RuntimeAssembly.nLoadFile(String path, Evidence evidence)
   at System.Reflection.Assembly.LoadFile(String path)
   at Duplicati.Library.Snapshots.WindowsSnapshot..ctor(String[] sourcepaths, Dictionary`2 options)
   at Duplicati.Library.Snapshots.SnapshotUtility.CreateSnapshot(String[] folders, Dictionary`2 options)
   at Duplicati.Library.Main.Operation.BackupHandler.GetSnapshot(String[] sources, Options options, ILogWriter log) => The system cannot find the file specified. (Exception from HRESULT: 0x80070002)]_

In Windows 10, .NET is updated to version 4.6.x. Perhaps the AlphaFS 451 libraries are not compatible with Windows 10 and .NET 4.6.x.   It might be theoretically possible to revert to .NET 4.5.1, however Windows10 updates will automatically update .NET.

**Comment by mnaiman** [comment]
@NickPyz Really youre mixing 2 problem together. Fix was for problem with AlphaFS not for AlphaVSS.
If you want working AlphaVSS copy old directory alphavss to your new installation. It will work.

**Comment by NickPyz** [comment]
@mnaiman  I am confused ...

The problem I am trying to solve was described when I first opened this thread - the warning thrown pointed to a problem with AlphaFS 2.0.0.0. At that point in time I was using vanilla duplicati 2.0.1.18 without having made any file / folder edits. 

Let's go back to that starting point and please walk me thru what you suggest to be the correct solution.

Thanks!

**Comment by mnaiman** [comment]
@NickPyz At first post you wrote that you have this error right? "Warnings: [Failed to create a snapshot: System.IO.FileLoadException: Could not load file or assembly 'AlphaFS, Version=2.0.0.0"

For this problem solution is: Problem is not in AlphaVSS but in Alpha FS. There is some mismatch in dependencies, because Duplicati installs 1.5 but is dependant on 2.0 AlphaFS from here https://github.com/alphaleonis/AlphaFS/releases

It worked for me and @mr-flibble 

Later you wrote that you have this error: Warnings: [Failed to create a snapshot: System.IO.FileNotFoundException: The system cannot find the file specified. (Exception from HRESULT: 0x80070002)

For that solution is to copy alphavss folder from previous install.

If it is still not working mention what warning or error Duplicati reporting

**Comment by NickPyz** [comment]
Solved! Starting from vanilla 2.0.1.18, the solution was 2 steps:

(1)copy missing alphavss folder from previous version 
**plus** 
(2) updating the AlphaFS to 2.0.0.0

Thanks.

**Comment by mr-flibble** [comment]
@NickPyz For me - I dit not need copy alphavss folder from previous version.

**Comment by Pablo2048** [comment]
Windows 7 x64 Pro. Same problem as @NickPyz here. Solved same way. Thanks.

**Comment by ovizii** [comment]
I also saw this after the upgrade:

`Last successful run:Today at 7:01 AM (took 00:01:30)
Next scheduled run:Tomorrow at 7:00 AM
Source:0 bytes
Backup:7.77 GB`

**Comment by GunnerCL** [comment]
Hi
I have the same problem, not only have previous installed version 2.0.1.18,

Copy the files

AlphaFS.dll
AlphaFS.pdb
AlphaFS.xml

In the root folder Duplicati 2, please I need urgent help to recover files from my backup.

**Comment by vb0** [comment]
I'm having the same, wasted the whole day with it. W10x64 won't see any files/folders, either when doing backups or even when trying to expand drives in the browser.

**Comment by GunnerCL** [comment]
This is a critical software, please need someone to help us solve the problem, we need to restore our files

**Comment by CyBiS** [comment]
@GunnerCL: if the solutions provided by @mnaiman, @mr-flibble and @NickPyz didn't work for you, simply roll back to the previous version [v2.0.1.17-2.0.1.17_canary_2016-05-14](https://github.com/duplicati/duplicati/releases/tag/v2.0.1.17-2.0.1.17_canary_2016-05-14). Before rolling back, you might want to export your backup configurations first though, just in case.

**Comment by GunnerCL** [comment]
I have already uninstalled 2 chances, I install the version Duplicati-2.0.1.17_canary_2016-05-17-x64.msi, but when running says it is the version "You are Currently running Duplicati - 2.0.1.18_canary_2016-08-10".

As I do to remove it completely?

**Comment by mbrum** [comment]
@GunnarCL 
Take a look at \User\appdata\local\duplicati 
Delete the folder 'update' if existing.

**Comment by GunnerCL** [comment]
does not work, I had removed before installing

**Comment by vb0** [comment]
Going on a tangent funny thing is that I caught this because I was always getting error on accessing "System Volume Information" (yes, I couldn't figure out how to make a GUI filter that matches that!) and now out of the blue all backups have no complaints!

I assume there are many others affected but just don't know it because everything looks fine even if they're doing a backup of literally nothing.

**Comment by kenkendk** [comment]
Thanks for reporting and helping out.

I am currently super busy, but I hope I can fix this and make a new build next week.

**Comment by schlm3** [comment]
@GunnerCL: If this is critical software for you, you should probably buy a backup software. duplicati 2 is still in beta and the update channel you are obviously using is "canary" which is aimed - as far as I know - to very early releases. So please use at your own risk...

**Comment by kenkendk** [comment]
All: thanks for reporting!

I have now found the problem and uploaded a new 2.0.1.19 build that has AlphaFS 2.0 (and some fixes to AlphaVSS as well).

I assume this means that this issue is now fixed.

@GunnerCL: Run the `Duplicati.Library.AutoUpdater.exe` tool to see where updates are stored, and then remove the update.

**Comment by kenkendk** [comment]
FYI: AlphaVSS was not fixed in 2.0.1.19, new build coming soon
